### PR TITLE
HBASE-28753 FNFE may occur when accessing the region.jsp of the replica region

### PR DESCRIPTION
**Pull request by guluo2016**
Details see: [HBASE-28753](https://issues.apache.org/jira/browse/HBASE-28753)

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 27s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 23s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 28s |  |  master passed  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  3s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 26s |  |  the patch passed  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 215m 20s |  |  hbase-server in the patch passed.  |
|  |   | 227m 12s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6117/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6117 |
| Optional Tests | javac javadoc unit |
| uname | Linux 9b3909b7d665 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / bf0f90fedd1ec3751ed0cade2ab10afa21649d66 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6117/1/testReport/ |
| Max. process+thread count | 4990 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6117/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Review by Apache9 — APPROVED** [review]

**Comment by guluo2016** [comment]
Hi @Apache9 , I also found that type conversion is no need in region.jsp, as fallow.
```
List<? extends Store> stores = region.getStores();
for (Store store : stores) {
   String cf = store.getColumnFamilyName();
   Collection<? extends StoreFile> storeFiles = store.getStorefiles(); %>
```

Maybe we can also fix it in this PR ,  Or we can open a new PR ?

File: hbase-server/src/main/resources/hbase-webapps/regionserver/region.jsp [hunk 1]
```diff
@@ -79,17 +86,20 @@
            <th>Len Of Biggest Cell</th>
            <th>Key Of Biggest Cell</th>
          </tr>
-       <%   for(StoreFile sf : storeFiles) { %>
+       <% int count = 0;
+          for(StoreFile sf : storeFiles) {
+            if (!fs.exists(sf.getPath())) continue;
```
Comment (pankaj72981) [inline hbase-server/src/main/resources/hbase-webapps/regionserver/region.jsp:91]: Do we need to check the file existence for primary region also?

Comment (guluo2016) [reply hbase-server/src/main/resources/hbase-webapps/regionserver/region.jsp:91]: Actually, we don't need to perform this check for primary region beacause it is authoritative.

But compared to the following code.
```
if (region.getRegionInfo().getReplicaId() > RegionInfo.DEFAULT_REPLICA_ID && !fs.exists(sf.getPath()))
```
I think the former may be more concise.

Comment (pankaj72981) [reply hbase-server/src/main/resources/hbase-webapps/regionserver/region.jsp:91]: IMO it's worth to add a primary region check, we will save unnecessary RPCs to Namenode.

Comment (guluo2016) [reply hbase-server/src/main/resources/hbase-webapps/regionserver/region.jsp:91]: Make sense, I will update it, thanks.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 39s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  5s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 45s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 51s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  spotless  |   0m 45s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 12s |  |  The patch does not generate ASF License warnings.  |
|  |   |   9m 28s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6117/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6117 |
| Optional Tests | dupname asflicense javac codespell detsecrets spotless |
| uname | Linux 3d4559709192 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 084cc4b1b12f1c5cb404f974cb1301d3339a5348 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 83 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6117/2/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 28s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 23s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 26s |  |  master passed  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  5s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 27s |  |  the patch passed  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 216m  8s |  |  hbase-server in the patch passed.  |
|  |   | 227m 58s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6117/2/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6117 |
| Optional Tests | javac javadoc unit |
| uname | Linux 5419c8db0a30 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 084cc4b1b12f1c5cb404f974cb1301d3339a5348 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6117/2/testReport/ |
| Max. process+thread count | 5258 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6117/2/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
@pankaj72981 Do you have any other concerns?

Thanks.

**Comment by pankaj72981** [comment]
> @pankaj72981 Do you have any other concerns?
> 
> Thanks.

No more suggestions, we are good to go now. Thanks @guluo2016

**Review by pankaj72981 — APPROVED** [review]
+1 LGTM

**Pull request merged**